easybuild/easyblocks/generic/bundle.py

```diff
@@ -324,6 +326,41 @@ def install_step(self):
             # close log for this component
             comp.close_log()
 
+    def make_module_req_guess(self):
+        """
+        Set module requirements from all comppnents, e.g. $PATH, etc.
+        During the install step, we only set these requirements temporarily.
+        Later on when building the module, those paths are not considered.
+        Therefore, iterate through all the components again and gather
+        the requirements.
+
+        Do not remove duplicates or check for existance of folders,
+        as this is done in the generic EasyBlock while creating
+        the modulefile already.
+        """
+        # Start with the paths from the generic EasyBlock.
+        # If not added here, they might be missing entirely and fail sanity checks.
+        final_reqs = super(Bundle, self).make_module_req_guess()
+
+        for cfg, comp in zip(self.comp_cfgs, self.comp_blocks):
+            self.log.info("Gathering module paths for component %s v%s", cfg['name'], cfg['version'])
+            reqs = comp.make_module_req_guess()
+
+            if not reqs or not isinstance(reqs, dict):
+                self.log.warning("Expected requirements to be a dict but is not. Therefore, this component is skipped.")
+                continue
+
+            for key, value in sorted(reqs.items()):
+                if isinstance(reqs, string_type):
+                    self.log.warning("Hoisting string value %s into a list before iterating over it", value)
+                    value = [value]
+                if key not in final_reqs:
+                    final_reqs[key] = value
+                else:
+                    final_reqs[key] += value
```

Comment:
The first `if` checks a situation that should almost never happen, then it is more efficient to use a `try..except` there.

And the update of the keys can be simplified with a `setdefault`.

```suggestion
            try:
                for key, value in sorted(reqs.items()):
                    if isinstance(reqs, string_type):
                        self.log.warning("Hoisting string value %s into a list before iterating over it", value)
                        value = [value]
                    
                    final_reqs.setdefault(key, [])
                    final_reqs[key] += value
            except AttributeError:
                self.log.warning("Expected requirements to be a dict but is not. Therefore, this component is skipped.")
```

Reply:
One more thing, `make_module_req_guess` should always return a dict, otherwise it there is an error somewhere. So instead of ignoring the component we should stop and raise an error.
And there is no need to print a warning for hoisting the value here. It's a minor change.

Reply:
Applied the suggestion

Reply:
> One more thing, make_module_req_guess should always return a dict, otherwise it there is an error somewhere. So instead of ignoring the component we should stop and raise an error.

> And there is no need to print a warning for hoisting the value here. It's a minor internal change.

Will add 👍 
For the warning, I basically just took it from the generic EasyBlock where the same thing is done. However, I agree that a waning is not necessary.